src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -1317,90 +1422,25 @@ export class StudyViewPageStore {
-            })
-            return [this.queriedSampleIdentifiers, this.defaultVisibleAttributes, ...promises]
-        },
-        invoke: async () => {
-            let studyViewFilter:StudyViewFilter = { studyIds: this.queriedPhysicalStudyIds.result } as any
-            return _.reduce(this.defaultVisibleAttributes.result.filter(c => c.datatype === ClinicalAttributeDataTypeConstants.STRING), (acc, next) => {
-                const uniqueKey = getClinicalAttributeUniqueKey(next);
-                acc[uniqueKey] = this.studyViewClinicalDataCountsCache
-                    .get({ attribute: next, filters: studyViewFilter })
-                    .result!;
-                return acc;
-            }, {} as any);
-        },
-        default: {},
-        onResult: (result) => {
-            _.forEach(result, (obj, uniqueKey) => {
-                //TODO: this is temporary. will be updated in next phase
-                if (obj.length < 2 && !_.includes(ALWAYS_SHOWN_ATTRS, uniqueKey)) {
-                    this.changeChartVisibility(uniqueKey, false);
-                } else {
-                    this.changeChartVisibility(uniqueKey, true);
-                }
-                if (obj.length > PIE_TO_TABLE_LIMIT) {
-                    this.chartsType.set(uniqueKey, ChartTypeEnum.TABLE);
-                }
-                if (this.chartsType.get(uniqueKey) === ChartTypeEnum.TABLE) {
-                    this.chartsDimension.set(uniqueKey, this.getTableDimensionByNumberOfRecords(obj.length));
-                }
-            });
-        }
-    });
-
-    readonly initialClinicalDataBins = remoteData<{ [id: string]: DataBin[] }>({
-        await: () => {
-            const promises = this.defaultVisibleAttributes.result.filter(c => c.datatype === ClinicalAttributeDataTypeConstants.NUMBER).map(attribute => {
-                return this.getDefaultInitialDataBinCounts(attribute);
-            });
-
-            return [this.defaultVisibleAttributes, ...promises];
-        },
-        invoke: async () => {
-            return _.reduce(this.defaultVisibleAttributes.result.filter(c => c.datatype === ClinicalAttributeDataTypeConstants.NUMBER), (acc, next) => {
-                const uniqueKey = getClinicalAttributeUniqueKey(next);
-                acc[uniqueKey] = this.getDefaultInitialDataBinCounts(next).result!;
-                return acc;
-            }, {} as any);
-        },
-        default: {},
+        default: [],
         onResult: (result) => {
-            _.forEach(result, (obj, uniqueKey) => {
-                if (obj.length < 2) {
-                    this.changeChartVisibility(uniqueKey, false);
-                } else {
-                    this.changeChartVisibility(uniqueKey, true);
-                }
+            _.each(result, (attribute) => {
+                this.changeChartVisibility(getClinicalAttributeUniqueKey(attribute), true);
```

Comment:
gonna guess that changeChartVisibility is an "action" that updates observable.  If so, we will get as many redundant render cycles as items in result.   you should put the loop inside a method tagged with action to avoid this.